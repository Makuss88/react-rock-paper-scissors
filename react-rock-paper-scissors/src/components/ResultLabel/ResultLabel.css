.green {
    color: #2ecc71;
    font-weight: bold;
}

.red {
    color: #f34a58;
    font-weight: bold;
}

.bold {
    font-weight: bold;
}