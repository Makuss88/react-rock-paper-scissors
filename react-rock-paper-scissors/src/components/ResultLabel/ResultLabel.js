import React, { Component } from 'react';
import './ResultLabel.css';

class ResultLabel extends Component {

    getLabel = (winner) => {
        let label = {
            text: "Fight!",
            style: ""
        }
        switch(winner) {
            case 0:
                label = {
                    text: "Tied",
                    style: "bold"
                }
                break;
            case 1:
                label = {
                    text: "You win!",
                    style: "green"
                }
                break;
            case 2: 
                label = {
                    text: "Computer wins",
                    style: "red"
                }
                break;
            default:
                break;
        }
        return label;
      }

    render() {
        const label = this.getLabel(this.props.winner);
        return(
            <p className={ label.style }> { label.text } </p>
        );
    }
}
export default ResultLabel;