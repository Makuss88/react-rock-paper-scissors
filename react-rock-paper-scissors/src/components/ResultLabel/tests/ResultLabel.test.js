import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import ResultLabel from '../ResultLabel';

Enzyme.configure({ adapter: new Adapter() });

describe('ResultLabel component', () => {

    it('displays "You win!" message in green if the winner is the user (1)', () => {
        const wrapper = shallow(<ResultLabel winner={ 1 } />); 
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('displays "Computer wins" message in red if the winner is the computer (2)', () => {
        const wrapper = shallow(<ResultLabel winner={ 2 } />); 
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('displays "Tied" message in bold if tied (0)', () => {
        const wrapper = shallow(<ResultLabel winner={ 0 } />); 
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('displays "Fight!" message if the game didn\'t start', () => {
        const wrapper = shallow(<ResultLabel winner={ null } />); 
        expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
  