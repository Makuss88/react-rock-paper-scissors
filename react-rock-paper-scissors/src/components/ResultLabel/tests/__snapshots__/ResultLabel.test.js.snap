// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultLabel component displays "Computer wins" message in red if the winner is the computer (2) 1`] = `
<p
  className="red"
>
   
  Computer wins
   
</p>
`;

exports[`ResultLabel component displays "Fight!" message if the game didn't start 1`] = `
<p
  className=""
>
   
  Fight!
   
</p>
`;

exports[`ResultLabel component displays "Tied" message in bold if tied (0) 1`] = `
<p
  className="bold"
>
   
  Tied
   
</p>
`;

exports[`ResultLabel component displays "You win!" message in green if the winner is the user (1) 1`] = `
<p
  className="green"
>
   
  You win!
   
</p>
`;
